tensorflow_addons/image/dense_image_warp.py

```diff
@@ -189,7 +186,9 @@ def dense_image_warp(image, flow, name="dense_image_warp"):
       ValueError: if height < 2 or width < 2 or the inputs have the wrong
         number of dimensions.
     """
-    with tf.name_scope(name):
+    with tf.name_scope(name or "dense_image_warp"):
+        image = tf.convert_to_tensor(image)
```

Comment:
I'm not sure whether we need to use convert_to_tensor everywhere. cc @karmel @seanpmorgan can you give us some input?

Reply:
I would say that it matches the [pattern seen in core](https://github.com/tensorflow/tensorflow/blob/943a11faa9ec2ca06d9c0185a132c0c5b3e65b3f/tensorflow/python/ops/image_ops_impl.py#L322) so I think it's probably best to keep it unified.

Whether we should be throwing an error on non-tensors instead of just converting them I think is a fair question... but for now this is good IMO

Reply:
Yeah, I have no objection. Just want to figure out when to use it, and when not (or unnecessary).

Reply:
[From the style guide](https://www.tensorflow.org/community/contribute/code_style#python_operations): 
> Operations that take tensors as arguments should call convert_to_tensor to convert non-tensor inputs into tensors if they are using C++ operations. Note that the arguments are still described as a Tensor object of a specific dtype in the documentation.

Not sure if there's a reason there other than to avoid some unfriendly C++ crash?